drivers/hmis_external_apis/app/jobs/hmis_external_apis/publish_external_forms_job.rb

```diff
@@ -48,25 +48,23 @@ def update_cdeds(definition)
     missing = []
     definition.walk_definition_nodes do |node|
       cded_key = node['link_id']
-      attrs = nil
-      case node['type']
-      when 'STRING', 'CHOICE', 'BOOLEAN'
-        # treat everything as a string for now
-        attrs = {
-          owner_type: owner_type,
-          field_type: 'string',
-          key: cded_key,
-          label: cded_key.humanize,
-          repeats: false,
-          UserID: user_id,
-          data_source_id: data_source_id,
-          form_definition_identifier: definition.identifier,
-        }
-      end
-      if attrs
-        cded = cded_by_key[cded_key]
-        cded ? cded.update!(attrs) : missing.push(attrs)
-      end
+      next unless ['STRING', 'CHOICE', 'BOOLEAN', 'INTEGER', 'TEXT', 'DATE'].include?((node['type']))
+
+      field_type = node['type'].downcase || 'string'
+      field_type = 'string' if field_type == 'choice'
```

Comment:
This special-casing for choice types is messy. And, it's connected to [this](https://github.com/greenriver/hmis-frontend/pull/691/files#diff-736ca0dfcf22fbe66ce15749daa3f61b9b3dadfef7764d7fd3450f73009d4ed9R79-R88) messy special-casing in the frontend code.

Should we spend more time to find a neater way? For example, add a `value_choice` column to the CDED table?

Or should we just add some todos in the code for now?

Reply:
I think we should probably pause on this; it looks like Gig is [working on the same thing](https://github.com/greenriver/hmis-warehouse/pull/4103)